shared/workspace_repo/workspace.go

```diff
@@ -595,6 +600,11 @@ func (s *TFEWorkspace) GetCurrentRunStatus(ctx context.Context) string {
 }
 
 func (s *TFEWorkspace) GetCurrentRunUrl(ctx context.Context) string {
+	state, err := s.HasState(ctx)
+	if err != nil || !state {
+		return s.GetWorkspaceUrl()
```

Comment:
this kind of hides an error but I don't think we need to introduce another error.